daemon-tests/Cargo.toml

```diff
@@ -14,6 +14,7 @@ mockall = "0.11"
 mockall_derive = "0.11"
 model = { path = "../model" }
 otel-tests = { version = "0.1", default-features = false }
+portpicker = "0.1.1"
```

Comment:
❓ The assumption is that `portpicker` is more reliable with picking a port than `TcpListener`?

I guess we can only test this by merging it and observing the behavior, or do you think we can actually test this?

Reply:
as described in the commit message, the mechanism for choosing the port is different. `TcpListener` just tries to bind to a local socket, which will give back a free port from the OS (but could lead to a race condition, if 2 tests ask at the same time).
`portpicker` starts by randomly choosing ports and checking whether they're free, only falling back to asking the OS to provide a port if it can't guess it :)